lib/galaxy/webapps/galaxy/api/datasets.py

```diff
@@ -126,10 +126,7 @@ def show(self, trans, id, hda_ldda='hda', data_type=None, provider=None, **kwd):
         Displays information about and/or content of a dataset.
         """
         # Get dataset.
-        try:
-            dataset = self.get_hda_or_ldda(trans, hda_ldda=hda_ldda, dataset_id=id)
-        except Exception as e:
-            return str(e)
+        dataset = self.get_hda_or_ldda(trans, hda_ldda=hda_ldda, dataset_id=id)
```

Comment:
Can you also try replacing `legacy_expose_api_anonymous` and use `expose_api_anonymous` ?
I think that should work fine here.

Reply:
I don't think the way exceptions are dealt with in the try/except block below is how non-legacy API calls are supposed to work.
Can we postpone that to a 19.09 `enhancement` PR?

Reply:
sure